modules/openapi-generator/src/main/resources/scala-play-framework/caseClass.mustache

```diff
@@ -0,0 +1,18 @@
+case class {{classname}}({{#vars}}
+  {{&name}}: {{^required}}Option[{{/required}}{{#isEnum}}{{classname}}.{{enumName}}.Value{{/isEnum}}{{^isEnum}}{{dataType}}{{/isEnum}}{{^required}}]{{/required}}{{#hasMore}},{{/hasMore}}{{^hasMore}}{{#isExtensibleObject}},{{/isExtensibleObject}}{{/hasMore}}{{/vars}}
+{{#parent}}  {{#isExtensibleObject}}{{parentPropName}}{{/isExtensibleObject}}: {{parent}}
+{{/parent}})
+
+object {{classname}} {
+  implicit lazy val {{classVarName}}JsonFormat: Format[{{classname}}] = {{^isExtensibleObject}}Json.format[{{classname}}]{{/isExtensibleObject}}{{#isExtensibleObject}}{
+{{>extensibleObjectJsonFormat}}
+  }{{/isExtensibleObject}}{{#vars}}{{#isEnum}}
+
+  // noinspection TypeAnnotation
+  object {{enumName}} extends Enumeration { {{#allowableValues}}{{#values}}
+    val {{.}} = Value{{/values}}{{/allowableValues}}
+
+    type {{enumName}} = Value
+    implicit lazy val {{enumName}}JsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
```

Comment:
May want to import `Format` and other Json related types at the object level here and/or above line 7, otherwise this assumes they exist in the including mustache file.

Reply:
This is a partial that's only used in `model.mustache` which has this import. I split the model into three main partials (`arrayCaseClass.mustache`, `caseClass.mustache` and `caseObject.mustache`) in an attempt to keep it somewhat cleaner.